### PR TITLE
Initial RTD hook implementation

### PR DESCRIPTION
**Pull request by deejay1**
Initial implementation of the RTD hook for automatic doc builds.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,5 +15,7 @@ jobs:
           cache: 'npm'
       - run: npm ci
       - run: npm run lint
-      - run: npm test -- --coverage
-      - uses: codecov/codecov-action@v1
\ No newline at end of file
+      - run: |
+          unset GITHUB_REF
```
Comment (bgalek) [inline .github/workflows/ci.yml:19]: why unset GITHUB_REF?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+*.lcov
+
+# nyc test coverage
+.nyc_output
+
+# Grunt intermediate storage (https://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
+
+# Bower dependency directory (https://bower.io/)
+bower_components
```
Comment (bgalek) [inline .gitignore:36]: lol?:)

File: src/main.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import * as core from '@actions/core'
+import * as github from '@actions/github'
+import {HttpClient} from '@actions/http-client'
+import {ITypedResponse} from '@actions/http-client/interfaces'
+
+core.setSecret('rtdToken')
```
Comment (bgalek) [inline src/main.ts:6]: looks like debug left

File: src/main.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    core.info(
+      `Build·triggered:·${resp.result.build_triggered}·Versions:·${resp.result.versions}·Ref:·${branch}`
+    )
+  } else {
+    core.warning(`Failed to trigger build.`)
+  }
+}
+
+if (github.context.ref) {
+  const branch = github.context.ref.replace('refs/heads/', '')
```
Comment (bgalek) [inline src/main.ts:30]: this code depends on dispatched event (pr/push) support both or fail for PR ;)

**Review by bgalek — COMMENTED** [review]
a lot of fuss just to make a HTTP request ;) 
didn't see any ts features used, beside that i found only some side-stuff ;)

File: src/main.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  run(branch)
```
Comment (bgalek) [inline src/main.ts:31]: await missing, youre not waiting for your promise to complete, and not doing anything after (.then())

**Review by bgalek — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (deejay1) [reply .gitignore:36]: stock ;)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bgalek) [reply .github/workflows/ci.yml:19]: yes, this would be better than like, unseting stuff :D

File: .gitignore (hunk 2 above)
Comment (bgalek) [reply .gitignore:36]: looks like not used code :D

File: src/main.ts (hunk 5 above)
Comment (bgalek) [reply src/main.ts:31]: seems like bug to me, are you sure that your code will wait for the execution before turning down the container?

Comment (bgalek) [reply src/main.ts:31]: can you share this github action sample in ts? :) I've seen some that commit node_modules :)

File: src/main.ts (hunk 3 above)
Comment (bgalek) [reply src/main.ts:6]: any docs on that? never heard of this, I've assumed that github masks logs by values...

File: src/main.ts (hunk 4 above)
Comment (deejay1) [reply src/main.ts]: this is gone completely now

File: src/main.ts (hunk 3 above)
Comment (deejay1) [reply src/main.ts:5]: https://github.com/actions/toolkit/tree/main/packages/core#setting-a-secret

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,103 @@
...
+.next
+
+# nuxt.js build output
+.nuxt
+
+# vuepress build output
+.vuepress/dist
+
+# Serverless directories
+.serverless/
```
Comment (bgalek) [inline .gitignore:85]: a lot of unused rules ;) vupress, next ,nuxt etc. :P

File: src/notifier.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import {ITypedResponse} from '@actions/http-client/interfaces'
+
+interface RtdResponse {
+  build_triggered: boolean
+  project: string
+  versions: string[]
+}
+
+export async function notify(): Promise<void> {
+  const branch = github.context.ref.replace('refs/heads/', '')
```
Comment (bgalek) [inline src/notifier.ts:13]:
```
const branch = process.env['GITHUB_HEAD_REF'] || process.env['GITHUB_REF'].slice('refs/heads/'.length);
```
would remove the context dependency, also would work for PR event

File: src/notifier.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  const hookUrl = core.getInput('hookUrl')
```
Comment (bgalek) [inline src/notifier.ts:14]: you should assert if the input is falsy

File: src/notifier.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  const rtdToken = core.getInput('rtdToken')
```
Comment (bgalek) [inline src/notifier.ts:15]: you should assert if the input is falsy

File: src/notifier.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  const client = new HttpClient('rtd-github')
+  const data = {branches: branch, token: rtdToken}
+  const resp: ITypedResponse<RtdResponse> = await client.postJson(hookUrl, data)
+  if (resp.result) {
```
Comment (bgalek) [inline src/notifier.ts:19]: if rtd returns 200 with null body it will fail? is it ok or not?

File: action.yml [hunk 11]
```diff
@@ -1,6 +1,14 @@
...
+description: 'Action to push a generic webhook to a custom RTD installation'
+branding:
+  icon: 'book'
+inputs:
+  hookUrl:
+    required: true
+    description: URL for the Generic API incoming webhook
+  rtdToken:
+    required: false
+    description: Token for
```
Comment (bgalek) [inline action.yml:11]: token for what? :)

File: src/notifier.ts (hunk 7 above)
Comment (deejay1) [reply src/notifier.ts:13]: well I need to push the branch into rtd so I need it's name

File: src/notifier.ts (hunk 9 above)
Comment (deejay1) [reply src/notifier.ts:15]: added `{required: true}` which does the same ;)

File: .gitignore (hunk 6 above)
Comment (deejay1) [reply .gitignore:85]: cleaned up a bit more

File: src/notifier.ts (hunk 8 above)
Comment (deejay1) [reply src/notifier.ts:14]: used `{required: true}`

**Review by adowsky — APPROVED** [review]

File: src/main.ts (hunk 5 above)
Comment (bgalek) [reply src/main.ts]: await will block till end of the execution

**Pull request merged**